Modules/mathmodule.c

```diff
@@ -2016,6 +2016,71 @@ math_factorial(PyObject *module, PyObject *arg)
 }
 
 
+/*[clinic input]
+math.lcm
+
+    x as a: object
+    y as b: object
+    /
+least common multiple of x and y
+[clinic start generated code]*/
+
+static PyObject *
+math_lcm_impl(PyObject *module, PyObject *a, PyObject *b)
+/*[clinic end generated code: output=6f83fb6d671074ba input=bd41b785dc2a4ff1]*/
+{
+    PyObject *g, *m, *f, *ab;
+    
+    a = PyNumber_Index(a);
+        if (a == NULL) {
+            return NULL;
+    }
+    b = PyNumber_Index(b);
+    if (b == NULL) {
+        return NULL;
+    }
+    if (a == 0 && b == 0) {
```

Comment:
Unrelated to reference counting, but this is not the right way to check whether the integers `a` and `b` are zero. You're checking whether the *pointers* are zero, i.e., `NULL`. That'll never be true. We can fix the handling of zeros once the refcounting is sorted out.

Owned references at this point: `a` and `b`